### DISCUSSION TITLE
Error: undefined: There is no template at js/tmpl/login.ejs undefined

### DISCUSSION THREAD

**Discussion by peppies** [docker-library/rabbitmq#665] [Q&A]
RabbitMQ 3.12.4
Erlang 26.0.2
Nginx 1.23.2

Enabled Plugins:
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_web_dispatch
  amqp_client
  cowboy
  cowlib

Greetings,

I am trying to get RabbitMQ Management UI to work on a website URL:
http://mywebsite.com/rabbitmq/

However, I am receiving this error in an orange bar at the bottom of the page:
**undefined: There is no template at js/tmpl/login.ejs undefined**

I looked in Chrome's network and other files are loading, just not login.ejs. Here is a screenshot:

![rabbitmq-error](https://github.com/docker-library/rabbitmq/assets/2999583/4cf1e535-ae1b-49c1-9e0d-98cc07944568)

_The UI does work on my server IP address though:_
http://[server-ip-address]:15672/

I have added the following to my Nginx server block for the domain name, which is referenced in the RabbitMQ Management Docs - https://www.rabbitmq.com/management.html#http-api-proxy:

    location ~* /rabbitmq/api/(.*?)/(.*) {
      proxy_pass http://localhost:15672/api/$1/%2F/$2?$query_string;
    }

    location ~* /rabbitmq/(.*) {
      rewrite ^/rabbitmq/(.*)$ /$1 break;
      proxy_pass http://localhost:15672;
    }

Seems like it should be working, but it is not.

**Comment by lukebakken** [comment]
Please prefer "Discussions" for when you have questions like this. There is, as of now, no evidence of a bug. More than likely it is a problem with your nginx configuration.

Could you please provide a **complete** docker compose project that starts RabbitMQ and nginx exactly the same way you are, so I can see the issue? Right now you're asking me to set this up myself when, ideally, I would just clone a repo with your files, and run `docker compose up` to see the issue. That way I can focus on figuring out the issue rather than figuring out to reproduce it. Thank you.

**Reply by peppies** [reply]
I've never used Docker before. I had a small "mix" project with a bunch of dependencies for Phoenix, Elixir, Broadway, etc which uses RabbitMQ. Although I think I have to install RabbitMQ separately on my server first, outside the "mix" project.

I installed RabbitMQ on Ubuntu using the APT method. Nginx was already running, so I simply used "sudo systemctl restart rabbitmq-server" to manually get RabbitMQ running. I'm in the testing phase first, just trying manually turn on RabbitMQ, get it set up and working properly. It's working, but not on the website's /rabbitmq/ directory. As far as I know, RabbitMQ isn't working inside any project. Right now in testing, it's just RabbitMQ and Nginx all by itself.

**Comment by lukebakken** [comment]
@peppies - you originally opened your issue in `docker-library/rabbitmq`, which is why I thought this is Docker related. Please start a new discussion here:

https://github.com/rabbitmq/rabbitmq-server/discussions

Please include the following information:
* Ubuntu, RabbitMQ, Erlang and Nginx versions
* Transcripts of commands you have run to install software
* **FULL** RabbitMQ and Nginx configuration files. They _must_ be complete.
* How are you determining if RabbitMQ is "working"?

Again, your job is to make it easy for me to see the issue, at which point I can assist you. Team RabbitMQ is small and we do not have the time to spend reproducing issues. Thanks for understanding.

**Comment by michaelklishin** [comment]
Please clear your browser's cache and see if that makes any difference.
